pkg/dns/ip.go

```diff
@@ -2,13 +2,19 @@ package dns
 
 import (
 	"net"
+	"strings"
 
 	"github.com/Nordix/simple-ipam/pkg/ipam"
 )
 
 type IPAM interface {
 	AllocateIP() (string, error)
 	FreeIP(ip string) error
+	ReserveIP(ip string) error
+}
+
+func IsAddressAlreadyAllocated(err error) bool {
+	return err != nil && strings.HasPrefix(err.Error(), "Address already allocated")
```

Comment:
Hmm, can we have a custom ErrorType and compare against it here? Checking string prefix is probably not the best.

Reply:
We already have the same approach in `store.go` and here error comes from `Nordix/simple-ipam` library. The only thing we can do is to check it against string literal.